code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -252,19 +249,21 @@ var/VOX_AVAILABLE_VOICES = list(
 	if(!cancorruptvox())
 		vox_corrupted = FALSE // this is to stop a sudden malf/ion making the announcement corrupt if it was on previously.
 	var/freq = 1
+	var/z_level = istype(loc, /obj/machinery/power/apc) ? 1 : src.z
```

Comment:
instead, use
`!z ? loc.z : z` (it might even be good to include some sanity above like `if(!loc) return`, though nullspaced AI probably has bigger issues going on than a VOX runtime)
Even if the AI is only supposed to be able to shunt on Station_Z this may not be the case on some future gimmickstation